typespec-extension/src/code-model-builder.ts

```diff
@@ -1332,8 +1343,11 @@ export class CodeModelBuilder {
     let trackConvenienceApi = op.convenienceApi ?? false;
     if (resp.responses && resp.responses.length > 0 && resp.responses[0].body) {
       const responseBody = resp.responses[0].body;
+      const unknownResponseBody =
+        responseBody.contentTypes.length > 0 && !isKnownContentType(responseBody.contentTypes);
+
       bodyType = this.findResponseBody(responseBody.type);
-      if (bodyType.kind === "Scalar" && bodyType.name === "bytes") {
+      if (unknownResponseBody && bodyType.kind === "Scalar" && bodyType.name === "bytes") {
```

Comment:
If content-type is unknown, and body is bytes, we'd take it means binary stream.